### PR TITLE
fix(treeview): fix switching to Focus Mode inside `TreeView`

### PR DESCRIPTION
**Pull request by vitaliirumiantsev-cengage**
Issue: #1420

## What I did

Fixed switching to Focus Mode inside `TreeView`

## Checklist 
- [x] changeset has been added
- [x] Pull request description is descriptive
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

## How to test

Enable screenreader -> Go to the storybook -> TreeView -> open any example -> use arrow up and arrow down to navigate inside TreeView

### CODE REVIEW

File: packages/react-magma-dom/src/components/TreeView/TreeItem.tsx [hunk 1]
```diff
@@ -310,6 +310,14 @@ export const TreeItem = React.forwardRef<HTMLLIElement, TreeItemProps>(
           selectableType={selectable}
           selected={selectedItem}
           theme={theme}
+          tabIndex={itemToFocus === itemId ? 0 : -1}
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/TreeView/TreeItem.tsx:313]: The challenge with moving this to here is that it causes the focus to be on the entire parent+children elements:
image

We want to keep the focus on just the item:
image

Comment (vitaliirumiantsev-cengage) [reply packages/react-magma-dom/src/components/TreeView/TreeItem.tsx:318]: done

File: packages/react-magma-dom/src/components/TreeView/TreeItem.tsx [hunk 2]
```diff
@@ -310,6 +315,14 @@ export const TreeItem = React.forwardRef<HTMLLIElement, TreeItemProps>(
...
+          onKeyDown={handleKeyDown}
+          ref={ref}
```
[the remark this answers is not in the document]
Comment (vitaliirumiantsev-cengage) [reply packages/react-magma-dom/src/components/TreeView/TreeItem.tsx]: done

File: packages/react-magma-dom/src/components/TreeView/useTreeItem.ts [hunk 3]
```diff
@@ -147,8 +147,14 @@ export function useTreeItem(props: UseTreeItemProps, forwardedRef) {
   };
 
   const handleClick = (event, itemId) => {
+    const isChecked = checkedStatus === IndeterminateCheckboxStatus.checked;
+
+    if (selectable === TreeViewSelectable.single && isChecked) {
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (vitaliirumiantsev-cengage) [reply packages/react-magma-dom/src/components/TreeView/useTreeItem.ts:154]: done

**Review by silvalaura — APPROVED** [review]
🌟

**Pull request merged**